### PR TITLE
change reviver to be new-crit friendly

### PR DESCRIPTION
**Pull request by Couls**
## What Does This PR Do

Small rewrite of reviver code to be new crit friendly and a bit cleaner
## Why It's Good For The Game

reviver implant was literally useless because you didn't go unconscious in new crit for long, now that most races are on new crit it makes more sense to tweak the reviver implant so it's useful to them

## Changelog
:cl:
tweak: reviver implant now heals even when you're conscious
/:cl:

### CODE REVIEW

File: code/modules/surgery/organs/augments_internal.dm [hunk 1]
```diff
@@ -268,34 +268,33 @@
...
-					owner.adjustFireLoss(-1)
-					revive_cost += 20
-				if(prob(40) && owner.getToxLoss())
-					owner.adjustToxLoss(-1)
-					revive_cost += 50
+		if(owner.health <= HEALTH_THRESHOLD_CRIT)
+			addtimer(CALLBACK(src, .proc/heal), 30)
 		else
-			cooldown = revive_cost + world.time
 			reviving = 0
```
Comment (farie82) [inline code/modules/surgery/organs/augments_internal.dm]: FALSE

File: code/modules/surgery/organs/augments_internal.dm [hunk 2]
```diff
@@ -268,34 +268,33 @@
...
-	if(cooldown > world.time)
-		return
-	if(owner.stat != UNCONSCIOUS)
-		return
-	if(owner.suiciding)
-		return
+			return
+	cooldown = revive_cost + world.time
 	revive_cost = 0
 	reviving = 1
```
Comment (farie82) [inline code/modules/surgery/organs/augments_internal.dm]: TRUE

Comment (Couls) [reply code/modules/surgery/organs/augments_internal.dm]: 🤮

File: code/modules/surgery/organs/augments_internal.dm (hunk 1 above)
Comment (Couls) [reply code/modules/surgery/organs/augments_internal.dm]: 🤮

**Comment by Couls** [comment]
> <3

🤮🤮🤮

**Comment by Fox-McCloud** [comment]
Might have to buff it even more, with some testing, but this is a great start.

**Comment by Couls** [comment]
> Might have to buff it even more, with some testing, but this is a great start.

I agree, but I wanted to atomize it. I might make a follow-up with 5% to inject 1 u of saline or epi with MUCH higher cooldown like 30 seconds

**Pull request merged**